rust/kernel/str.rs

```diff
@@ -89,8 +89,13 @@ impl CStr {
     /// must not be mutated.
     #[inline]
     pub unsafe fn from_char_ptr<'a>(ptr: *const c_types::c_char) -> &'a Self {
-        let len = bindings::strlen(ptr) + 1;
-        Self::from_bytes_with_nul_unchecked(core::slice::from_raw_parts(ptr as _, len as _))
+        // SAFETY: `ptr` is a valid pointer to a `NUL`-terminated C string.
```

Comment:
Do we need to mention that this property (and the ones just below) is guaranteed by the `/// # Safety` function prerequisite? Not sure how to formulate.

Reply:
Yeah, `// SAFETY` comments should not just say "what", but "why" too, i.e. something like:

```suggestion
        // SAFETY: The safety precondition guarantees `ptr` is a valid pointer to a `NUL`-terminated C string.
```